### PR TITLE
API Integration and Hide API Keys

### PR DESCRIPTION
**Pull request by lnabrigo18**

### CODE REVIEW

**Review by Eunicegenel — APPROVED** [review]

File: app/models/broker_stock.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class BrokerStock < ApplicationRecord
+  validates :user_id, presence: true
+  validates :stock_id, presence: true
```
Comment (analyn-cajocson) [inline app/models/broker_stock.rb]: you don't have to validate the `foreign_keys`

File: app/models/broker_stock.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
...
+  validates :company, presence: true
+  validates :ticker, presence: true
+  validates :price, presence: true
+  validates :created_at, presence: true
```
Comment (analyn-cajocson) [inline app/models/broker_stock.rb]: Rails always gives `created_at` a value. You don't have to validate this attribute

File: app/models/buyer_stock.rb [hunk 3]
```diff
@@ -1,2 +1,10 @@
 class BuyerStock < ApplicationRecord
+  validates :user_id, presence: true
+  validates :stock_id, presence: true
+  validates :total_price, presence: true
+  validates :quantity, presence: true
+  validates :company, presence: true
+  validates :ticker, presence: true
+  validates :price, presence: true
+  validates :created_at, presence: true
```
Comment (analyn-cajocson) [inline app/models/buyer_stock.rb]: same comment as above

File: app/models/stock.rb [hunk 4]
```diff
@@ -1,2 +1,12 @@
 class Stock < ApplicationRecord
+  validates :company, presence: true
+  validates :ticker, presence: true
+  validates :price, presence: true
+  validates :created_at, presence: true
```
Comment (analyn-cajocson) [inline app/models/stock.rb]: same comment as above

File: app/models/transaction.rb [hunk 5]
```diff
@@ -1,2 +1,11 @@
 class Transaction < ApplicationRecord
+  validates :user_id, presence: true
+  validates :stock_id, presence: true
+  validates :broker_id, presence: true
+  validates :total_price, presence: true
+  validates :quantity, presence: true
+  validates :company, presence: true
+  validates :ticker, presence: true
+  validates :price, presence: true
+  validates :created_at, presence: true
```
Comment (analyn-cajocson) [inline app/models/transaction.rb]: same comment as above

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]
Kindly include the `Gemfile.lock` to this PR

**Review by Eunicegenel — APPROVED** [review]

**Review by analyn-cajocson — APPROVED** [review]

**Pull request merged**